dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionHelper.java

```diff
@@ -60,29 +60,27 @@ public void evictAndSummarize(Collection collection) {
         // Ensure that entries is empty
         // This is probably unnecessary
         collection.setEntries(new HashSet<>());
-        collection.setWorkflowsLength(entryDAO.getWorkflowsLength(collection.getId()));
+        collection.setWorkflowsLength(entryDAO.getBioWorkflowsLength(collection.getId()) + entryDAO.getAppToolsLength(collection.getId()));
         collection.setToolsLength(entryDAO.getToolsLength(collection.getId()));
+        collection.setNotebooksLength(entryDAO.getNotebooksLength(collection.getId()));
     }
 
     public void evictAndSummarize(java.util.Collection<? extends Collection> c) {
         c.forEach(this::evictAndSummarize);
     }
 
     public void evictAndAddEntries(Collection collection) {
+        Long collectionId = collection.getId();
         Session currentSession = sessionFactory.getCurrentSession();
         currentSession.evict(collection);
-        List<CollectionEntry> collectionWorkflows = entryDAO.getCollectionWorkflows(collection.getId());
-        List<CollectionEntry> collectionServices = entryDAO.getCollectionServices(collection.getId());
+        List<CollectionEntry> collectionWorkflows = entryDAO.getAllCollectionWorkflows(collectionId);
```

Comment:
This gets all app tools, bioworkflows, notebooks and services. On line 113 - 118, you also get the individual lists for each type.

You could avoid calling `getAllCollectionWorkflows` and instead get each individual type and add it to to `collectionEntries` here. This way, you also have the lists of each individual type to calculate the count later.